lib/vmpooler/vsphere_helper.rb

```diff
@@ -196,31 +196,62 @@ def find_folder(foldername)
       base
     end
 
+    def get_host_utilization(host, model=nil, limit=90)
+      if model and not host.hardware.cpuPkg[0].description.split().include? model
+        return nil
+      end
+      if not host.runtime.inMaintenanceMode
+        if host.overallStatus == 'green'
+          cpu_usage = host.summary.quickStats.overallCpuUsage
+          cpu_size = host.summary.hardware.cpuMhz * host.summary.hardware.numCpuCores
+          cpu_utilization = (cpu_usage.to_f / cpu_size.to_f) * 100
+          memory_usage = host.summary.quickStats.overallMemoryUsage
+          memory_size = host.summary.hardware.memorySize / 1024 / 1024
+          memory_utilization = (memory_usage.to_f / memory_size.to_f) * 100
+          if not cpu_utilization > limit and not memory_utilization > limit
+            utilization = cpu_utilization + memory_utilization
+            return [ utilization, host ]
+          end
+        end
+      end
+      return nil
+    end
+
```

Comment:
I ran through and put together something that looks nearly like this and still does the same thing. I think it captures what you were suggesting and helps clean it up. Thank you.